src/trust.rs

```diff
@@ -0,0 +1,157 @@
+use base64::encode;
+use chrono::{Duration, Utc};
+use rcgen::{
+    BasicConstraints, Certificate, CertificateParams, DnType, ExtendedKeyUsagePurpose, IsCa,
+    KeyIdMethod, KeyPair,
+};
+use serde_json::{json, Value};
+use std::fs::File;
+use std::io::Write;
+use std::{fs, process::exit, str::from_utf8};
+
+#[allow(non_camel_case_types)]
+enum CertificateType {
+    app,
+    device,
+}
+
+#[allow(non_camel_case_types)]
+struct cert {
+    certificate: Certificate,
+}
+
+impl cert {
```

Comment:
Yeah, so the reason behind it is when we generate it using the `generate_certificate` function and then do 
`certificate.serialize_pem()` and `certificate.serialize_private_key_pem()`, we get an error that the value of the certificate was moved earlier. To avoid this I created a struct, and as we have a struct we can move repetitive parameters to it.

Reply:
You already use a reference to the certificate, it works fine without the struct : 
https://gist.github.com/jbtrystram/6e4645b4d22e598b6f768f4b0593d6fc

Reply:
Thanks!